### ISSUE TITLE
pipeline crash in rule combine_chrom_bams

### ISSUE THREAD

**Issue by irepansalvador** [nanoporetech/sockeye#15]
Hi, many thanks for sockeye.
I have been trying to run the pipeline and it crashes when trying to execute rule `combine_chrom_bams`
This is the output (I changed the actual local paths for "PATH"):

```
Activating conda environment: .snakemake/conda/0f1b2ad4609c701c53f858078e3a5483
2022-07-20 10:35:10 -- Writing chrom-specfic BAM files to /PATH/outs/run1/demux/chroms
100%|███████████████████████████████| 41/41 [00:01<00:00, 22.03it/s]
[Wed Jul 20 10:35:12 2022]
Finished job 9.
5 of 18 steps (28%) done
MissingInputException in line 293 of /PATH/sockeye/rules/process_bams.smk:
Missing input files for rule combine_chrom_bams:
    output: /PATH/outs/run1/demux/tagged.sorted.bam, /PATH/outs/run1/demux/tagged.sorted.bam.bai
    wildcards: run_id=run1
    affected files:
        /PATH/outs/run1/demux/chroms/geneA.tagged.bam
```
I understand that the input of this rule is the function ` gather_chrom_bams`, that requires the files `CHROM_BAM_FULLY_TAGGED`. Checking in the 'chroms' folder I just see `{chrom}.sorted.bam` so I conclude that the rule(s) that produce the `{chrom}.tagged.bam` files are not being executed (eg `add_gene_tags_to_bam`)

In the initialization of the pipeline I don't see the `add_gene_tags_to_bam` so I wonder if this is related to the problem:
```

Job stats:
job                              count    min threads    max threads
-----------------------------  -------  -------------  -------------
align_to_ref                         1              7              7
all                                  1              1              1
call_cat_fastq                       1              7              7
call_paftools                        1              1              1
cleanup_headers_1                    1              1              1
combine_adapter_tables               1              1              1
combine_chrom_bams                   1              1              1
combine_stranded_fastqs              1              1              1
construct_expression_matrix          1              1              1
count_cell_gene_umi_reads            1              1              1
cp_batch_fastqs                      1              1              1
extract_barcodes                     1              7              7
get_chrom_sizes                      1              1              1
process_expression_matrix            1              1              1
split_bam_by_chroms                  1              7              7
summarize_adapter_table              1              1              1
umap_plot_genes                      4              1              1
umap_plot_mito_genes                 1              1              1
umap_plot_total_umis                 1              1              1
umap_reduce_expression_matrix        1              1              1
umi_gene_saturation                  1              1              1
total                               24              1              7


This was a dry-run (flag -n). The order of jobs does not reflect the order of execution.
```

Thanks in advance.
Irepan

**Comment by jbeaulaurier** [comment]
Hi Irepan,
I've encountered some issues with this step in the past after a job is interrupted and I try to pick up where it left off. If you try deleting all the output and starting fresh (or specifying a different output directory), does it make it past this problem?
Thanks,
John

**Comment by irepansalvador** [comment]
Hi John,

Thank you for your prompt response.
I have tried both deleting the OUTPUT folder and specifying a new one and it crashes on the same point. Also I have just tried to delete the environment and repo to start from scratch the installation and it crashes again.
I wonder if there is a toy dataset that I could try the pipeline with it, just to make sure is not related with the input files?

thanks,
Irepan

**Comment by jbeaulaurier** [comment]
Hi Irepan,
I'll be adding some test data very soon and will let you know when that is complete. Hopefully that will help pinpoint the problem.
Best,
John

**Comment by irepansalvador** [comment]
Hi John,

Thank you again.
Im looking forward to try it with the test data.

best,
Irepan

**Comment by jbeaulaurier** [comment]
Hi Irepan,

I've just added some test data and a test script to the ``master`` branch. Go ahead and do a fresh ``git pull`` to get it all and check out the updated README for instructions on running the test. Please let me know how it goes and we'll go from there.

Thanks,
John

**Comment by irepansalvador** [comment]
Hi John,

Many thanks for this, I will do the test next week when I'm back from a short holidays break.
I will keep you updated.

best
Irepan

**Comment by irepansalvador** [comment]
Hi John,

I have run the test script you kindly provided and it ran without any problem, which pointed to a problem with the input files I had.
I identified that the the source of the error was my gtf file. The genes I want to recover are from a transgene construct, so I needed to provide a manually curated gtf (used before to map reads using 10x cellranger).
It seems that the gtf file didnt have the correct format/annotations and it was crashing because of this. Two specific problems I identified are:
- The feature names had an "_" on them which created problems when processing the files
- The transgenes were annotated only as transcripts which created problems in the assig_genes.py step

By removing "_" from the feature names and modifying the gtf file the pipeline now works.

I'm looking forward to continue using sockeye, thank you again!

Irepan
